doc/tutorials/platformer/step_5.md

```diff
@@ -0,0 +1,318 @@
+# 5. Controlling Movement
+
+If you were waiting for some serious coding, this chapter is it.  Prepare yourself as we dive in!
+
+
+## Keyboard Controls
+
+The first step will to be allow control of Ember via the keyboard.  We need to start by adding the
+appropriate mixins to the game class and Ember. Add the following:
+
+`lib/ember_quest.dart'
+
+```dart
+import 'package:flame/events.dart';
+
+class EmberQuestGame extends FlameGame with HasKeyboardHandlerComponents {
+```
+
+`lib/actors/ember.dart`
+
+```dart
+class EmberPlayer extends SpriteAnimationComponent
+    with KeyboardHandler, HasGameRef<EmberQuestGame> {
+```
+
+Now we can add a new method:
+
+```dart
+  @override
+  bool onKeyEvent(RawKeyEvent event, Set<LogicalKeyboardKey> keysPressed) {
+    return true;
+  }
+```
+
+Like before, if this did not trigger an auto-import, you will need the following:
+
+```dart
+import 'package:flutter/services.dart';
+```
+
+To control Ember's movement, it is easiest to set a variable where we think of the direction of
+movement like a normalized vector, meaning the value will be restricted to -1, 0, or 1.  So let's
+set a variable at the top of the class:
+
+```dart
+  int horizontalDirection = 0;
+```
+
+Now in our `onKeyEvent` method, we can register the key pressed by adding:
+
+```dart
+@override
+  bool onKeyEvent(RawKeyEvent event, Set<LogicalKeyboardKey> keysPressed) {
+    horizontalDirection = 0;
+    horizontalDirection += (keysPressed.contains(LogicalKeyboardKey.keyA) ||
+            keysPressed.contains(LogicalKeyboardKey.arrowLeft))
+        ? -1
+        : 0;
+    horizontalDirection += (keysPressed.contains(LogicalKeyboardKey.keyD) ||
+            keysPressed.contains(LogicalKeyboardKey.arrowRight))
+        ? 1
+        : 0;
+
+    return true;
+  }
+```
+
+Let's make Ember move by adding a few lines of code and creating our `update` method.  First, we
+need to define a velocity variable for Ember.  Add the following at the top of your class:
+
+```dart
+final Vector2 velocity = Vector2.zero();
+final double moveSpeed = 200;
+```
+
+This establishes a base velocity of 0 and stores `moveSpeed` so we can adjust as necessary to suit
+how the game-play should be. Next, add the `update` method with the following:
+
+
+```dart
+  @override
+  void update(double dt) {
+    velocity.x = horizontalDirection * moveSpeed;
+    position += velocity * dt;
+    super.update(dt);
+  }
+```
+
+If you run the game now, Ember moves left and right using the arrow keys or the `A` and `D` keys.
+You may have noticed that Ember doesn't look back if you are going left, to fix that, add the
+following code at the end of your `update` method:
+
+```dart
+if (horizontalDirection < 0 && scale.x > 0) {
+    flipHorizontallyAroundCenter();
```

Comment:
Since the anchor is in the center, you can simply use `flipHorizontally()` here. This would also work better if you decide to shift the anchor off-center.

Reply:
Fixed.  In the beginning I was anchor `bottomLeft` when I wrote this, but then changed to `center` because I thought I was going to do something else, but then didn't.